src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,403 @@
+     * <ul>
+     * <li><strong>Bouncy Castle</strong>
+     * <ul>
+     * <li>{@link org.bouncycastle.asn1.cms.ContentInfo}
+     * <li>{@link org.bouncycastle.jce.spec.ECNamedCurveParameterSpec}
+     * <li>{@link org.bouncycastle.pkcs.PKCS10CertificationRequest}
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @CheckForNull
+    public Object getRawObject() {
+        return object;
+    }
+
+    /**
+     * Obtains the fingerprint of the private key in the "ab:cd:ef:...:12" format, which basically is an SHA1 digest
+     * from the key, encoded in hex format.
+     * 
+     * @return private key fingerprint in hex format "ab:cd:ef:...:12", null if the private key could not be obtained
+     * from the current PEM data.
+     * @throws IOException thrown if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPrivateKeyFingerprint() throws IOException {
+        PrivateKey key = toPrivateKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestSHA1(key));
+    }
+
+    /**
+     * Obtains the fingerprint of the public key in the "ab:cd:ef:...:12" format, which basically is an MD5 digest from
+     * the key, encoded in hex format.
+     * 
+     * @return public key fingerprint in hex format "ab:cd:ef:...:12", null if the public key could not be obtained from
+     * the current PEM data.
+     * @throws IOException if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPublicKeyFingerprint() throws IOException {
+        PublicKey key = toPublicKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestMD5(key));
+    }
+
+    /**
+     * Generates an SHA1 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestSHA1(@Nonnull Key k) throws IOException {
+        try {
+            return getKeyDigest(k, "SHA1");
+        } catch (NoSuchAlgorithmException e) {
+           throw new AssertionError("SHA1 algorithm not found to create digest");
+        }
+    }
+
+    /**
+     * Generates an MD5 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestMD5(@Nonnull Key k) throws IOException {
+        try {
+            return getKeyDigest(k, "MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError("SHA1 algorithm not found to create digest");
```

Comment:
This is MD5 and I would reference https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html where it says:

> Every implementation of the Java platform is required to support the following standard MessageDigest algorithms:
>
> * MD5
> * SHA-1
> * SHA-256